lib/pure/times.nim

```diff
@@ -1121,12 +1296,137 @@ proc evaluateStaticInterval(interval: TimeInterval): Duration =
     minutes = interval.minutes,
     hours = interval.hours)
 
+proc between*(start_dt, end_dt:DateTime, strict: bool = true): TimeInterval =
+  ## Evaluate difference between two dates in ``TimeInterval`` format, so, it
+  ## will be relative.
+  ##
+  ## ATTENTION! It's not recommended to use ``between`` for ``DateTime's`` in 
+  ## different ``TimeZone's``. If used in such manner with ``strict`` parameter 
+  ## set to ``true`` it will result in error.
+  ##
+  ## Set ``strict = false`` only if you know what you're doing! The results in
+  ## this case mey differ from that what you can expect!
+  runnableExamples:
+    var dt1 = parse("1970-03-04T03:00:00+00:00", "yyyy-MM-dd'T'HH:mm:sszzz")
```

Comment:
Please don't abuse runnableExamples for tests. What you have here is far too much.

Reply:
ok